flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/FlinkKubeClient.java

```diff
@@ -96,9 +98,10 @@
 	 * Watch the pods selected by labels and do the {@link PodCallbackHandler}.
 	 *
 	 * @param labels labels to filter the pods to watch
-	 * @param callbackHandler {@link PodCallbackHandler} will be called when the watcher receive the corresponding events.
+	 * @param podsWatcher watcher used to process the {@link PodCallbackHandler}
+	 * @return Return a watch for pods. It needs to be closed after use.
 	 */
-	void watchPodsAndDoCallback(Map<String, String> labels, PodCallbackHandler callbackHandler);
+	KubernetesWatch watchPodsAndDoCallback(Map<String, String> labels, KubernetesPodsWatcher podsWatcher);
```

Comment:
With this change, we make a fabric8 class part of the API since `KubernetesPodsWatcher` extends `Watcher<Pod>`. Why don't we add a `handleError` method to `PodCallBackHandler`?

Reply:
You are right. Adding the argument `KubernetesPodsWatcher` will break the abstraction of `FlinkKubeClient` API. I will move it to a new method `PodCallbackHandler#handleFatalError`.

Reply:
I think this still needs to be changed to
```suggestion
	KubernetesWatch watchPodsAndDoCallback(Map<String, String> labels, PodCallbackHandler podCallbackHandler);
```

Reply:
I get your concern and thanks a lot for the valid fixup.